src/tools/rbd_mirror/image_replayer/snapshot/Replayer.cc

```diff
@@ -1004,6 +1112,36 @@ void Replayer<I>::handle_create_non_primary_snapshot(int r) {
 
   dout(15) << "local_snap_id_end=" << m_local_snap_id_end << dendl;
 
+  create_group_snap_finish();
+}
+
+template <typename I>
+void Replayer<I>::create_group_snap_finish() {
+  if (!m_remote_mirror_snap_ns.group_spec.is_valid() ||
+      m_remote_mirror_snap_ns.group_snap_id.empty()) {
+    update_mirror_image_state();
+    return;
+  }
+
+  dout(10) << dendl;
+
+  auto ctx = create_context_callback<
+    Replayer<I>, &Replayer<I>::handle_create_group_snap_finish>(this);
+
+  m_replayer_listener->create_mirror_snapshot_finish(
```

Comment:
This sets the group snap to Complete but the non-primary .group snap on the image is still incomplete. That is set to complete only after the data is copied.